### ISSUE TITLE
[pkgs] better documentation & tests of `--dir` and `--file` corner cases

### ISSUE THREAD

**Issue by compleatang** [monax/monax#1390]
**Problem** -- we have some inconsistencies in how we are resolving a `--dir` and `--flag` corner case.

The following works when monax is run from a directory which is not the application's root directory (not the same as the `--dir` flag).

```
monax pkgs do -a $addr -c foo --dir $HOME/.monax/apps/my_app --file epm.yaml
```

The following does not work when monax is run from a directory which is not the application's root directory.

```
monax pkgs do -a $addr -c foo --dir $HOME/.monax/apps/my_app --file test.yaml
```

However, the following does work.

```
monax pkgs do -a $addr -c foo --dir $HOME/.monax/apps/my_app --file $HOME/.monax/apps/my_app/test.yaml
```

**Solution** -- there are two possible solutions here:

(a) better documentation that the `--file` flag requires an absolute path; or
(b) resolve the corner case correctly so that when `--dir` flag is given `--file` tries to resolve relative to the `--dir` flag rather than relative to `$pwd` and if there is no resolution then tries to resolve the absolute path. (b) is the resilient fix to the problem we should implement but (a) is a quick fix that should go into the next release.

**Comment by zramsay** [comment]
bumping to 0.18 because 0.17 doesn't have this feature implemented

**Comment by zramsay** [comment]
closed in #1429
